File: android-app/app/src/main/java/com/se1853_jv/labverse/data/utils/EncoderService.java
package com.se1853_jv.labverse.data.utils;

import android.os.Build;

import java.util.Arrays;
import java.util.Base64;

public class EncoderService {
    public static String encode(String str) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            return Arrays.toString(Base64.getUrlEncoder().encode(str.getBytes()));
        }
        return null;
    }

    public static String decode(String str) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            return Arrays.toString(Base64.getUrlDecoder().decode(str));
        }
        return null;
    }

}

File: services/paper-service/src/main/kotlin/com/se1853_jv/service/EncoderService.kt
package com.se1853_jv.service

import org.springframework.stereotype.Service
import java.util.*

@Service
class EncoderService {
    fun encode(str: String): String = Base64.getUrlEncoder().encodeToString(str.toByteArray())

    fun decode(str: String?): String = String(Base64.getUrlDecoder().decode(str))
}